File: 2018-12-31/1427/1427.java
import java.io.*;

class Main{
	private static String numbers;
	public static void main(String[] args) throws Exception{
		BufferedReader bfr = new BufferedReader(new InputStreamReader(System.in));
		numbers = bfr.readLine();
		StringBuilder strBuilder = new StringBuilder().append(numbers);
		quickSort(strBuilder);
		System.out.println(strBuilder);
	}

	private static void quickSort(StringBuilder str){
		quickSort(str, 0, numbers.length()-1);
	}

	private static void quickSort(StringBuilder str, int start, int end){
		int part2 = partition(str, start, end);

		if(start < part2-1)
			quickSort(str, start, part2-1);
		if(part2 < end)
			quickSort(str, part2, end);
	}

	public static int partition(StringBuilder str, int start, int end){
		char pivot = str.charAt((start+end)/2);

		while(start <= end){
			while(pivot < str.charAt(start))start++;
			while(pivot > str.charAt(end))end--;
			if(start<=end){
				swap(str, start, end);
				start++; end--;
			}
		}
		return start;
	}

	private static void swap(StringBuilder str, int start, int end){
		char temp = str.charAt(start);
		str.setCharAt(start, str.charAt(end));
		str.setCharAt(end, temp);
	}

}

File: 2018-12-31/1427/1427.c
#include <stdio.h>
#include <string.h>

void quickSort(char arr[], int start, int end);
int partition(char arr[], int start, int end);
void swap(char arr[], int start, int end);

int main(){
	char string[12];
	scanf("%s", string);
	int end = strlen(string);

	quickSort(string, 0, end-1);

	for(int i = 0; i < end; i++)
		printf("%c", string[i]);
}

void quickSort(char arr[], int start, int end){
	int part2 = partition(arr, start, end);
	if(start < part2-1)
		quickSort(arr, start, part2-1);
	if(part2 < end)
		quickSort(arr, part2, end);
}

int partition(char arr[], int start, int end){
	char pivot = arr[(start+end)/2];

	while(start<=end){
		while(arr[start] > pivot) start++;
		while(arr[end] < pivot) end--;
		if(start<=end){
			swap(arr, start, end);
			start++;
			end--;
		}
	}

	return start;
}

void swap(char arr[], int start, int end){
	char temp = arr[start];
	arr[start] = arr[end];
	arr[end] = temp;
}